heaps/min_heap.py

```diff
@@ -19,18 +22,32 @@ def __init__(self):
     def add(self, key, value = None):
         """ This method adds a HeapNode instance to the heap
             If value == None the new node's value should be set to key
-            Time Complexity: ?
-            Space Complexity: ?
+            Time Complexity: O(log n)
+            Space Complexity: O(1)
         """
-        pass
+        node = None
+        if value == None:
+            value = key
+        
+        node = HeapNode(key, value)
+        self.store.append(node)
+        self.heap_up(self.last_index())
+
 
     def remove(self):
```

Comment:
✨ However space complexity is O(log n) because of `heap_down`'s recursive call stack